### PR TITLE
Document discovery

### PR DESCRIPTION
**Pull request by dolfies**

### CODE REVIEW

File: pages/resources/discovery.mdx [hunk 1]
```diff
@@ -0,0 +1,443 @@
+# Discovery
+
+### Searching Discovery
+
+Discord utilizes [Algolia](https://www.algolia.com/) to power search for discovery. You can search for guilds by name, description, keywords, and more. Reference the [Algolia Search documentation](https://www.algolia.com/doc/rest-api/search/) for more information on how to search. A [proxied version of the Algolia Search API](#search-discoverable-guilds) is available, with limited parameter support.
+
+<Alert type="info">
+
+All hits returned from Algolia are [discoverable guild](#discoverable-guild-object) objects. However, some fields such as `features` are truncated to only include values that are relevant to discovery.
```
Comment (splatterxl) [inline pages/resources/discovery.mdx:9]: Specify the features that are relevant if known pls

File: pages/resources/discovery.mdx [hunk 2]
```diff
@@ -0,0 +1,443 @@
...
+
+</Alert>
+
+###### Algolia Credentials
+
+These credentials are for production only.
```
Comment (splatterxl) [inline pages/resources/discovery.mdx:15]:
```suggestion
These credentials are for Discord production’s Algolia client only. 
```

File: pages/resources/discovery.mdx [hunk 3]
```diff
@@ -0,0 +1,443 @@
...
+| approximate_presence_count | integer                                                       | Approximate number of online members in the guild                                                |
+| emojis? ^1^                | array[[emoji](/resources/emoji#emoji-object) object]          | Custom guild emojis; limited to 30 entries                                                       |
+| emoji_count? ^1^           | integer                                                       | Total number of custom guild emojis                                                              |
+| stickers? ^1^              | array[[sticker](/resources/sticker#sticker-object) object]    | Custom guild stickers; limited to 30 entries                                                     |
+| sticker_count? ^1^         | integer                                                       | Total number of custom guild stickers                                                            |
+| auto_removed               | boolean                                                       | Whether the guild has automatically been removed from discovery for not hitting required targets |
```
Comment (splatterxl) [inline pages/resources/discovery.mdx:48]: Why is this in a discoverable guild object? Shouldn’t guilds with this be removed from Algolia automatically?

File: pages/resources/discovery.mdx [hunk 4]
```diff
@@ -0,0 +1,443 @@
...
+    "WELCOME_SCREEN_ENABLED"
+  ],
+  "emojis": [],
+  "emoji_count": 339
+}
+```
+
+### Discovery Requirements Object
+
+Represents a guild's path towards meeting the requirements of joining discovery.
```
Comment (splatterxl) [inline pages/resources/discovery.mdx:122]:
```suggestion
Represents a guild's progress on meeting the requirements of joining discovery.
```

File: pages/resources/discovery.mdx [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+| minimum_size?                       | ?integer                                                                 | The minimum guild member count requirement                                                             |
+| grace_period_end_date?              | ISO8601 timestamp                                                        | When the guild's grace period ends                                                                     |
+
+^1^ Certain guilds, such as those that are [verified](https://discord.com/verification), are exempt from discovery requirements. These guilds will not have a fully populated discovery requirements object (guaranteed to receive only `sufficient` and `sufficient_without_grace_period`).
+
+###### Discovery NSFW Properties Structure
+
+| Field                        | Type                          | Description                                                    |
+| ---------------------------- | ----------------------------- | -------------------------------------------------------------- |
+| channels?                    | array[snowflake]              | The IDs of the channels with names containing disallowed terms |
```
Comment (splatterxl) [inline pages/resources/discovery.mdx:152]:
```suggestion
| channels?                    | array[snowflake]              | The channels with names containing disallowed terms |
```

File: pages/resources/discovery.mdx [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+  "keywords": ["test"],
+  "emoji_discoverability_enabled": true,
+  "partner_actioned_timestamp": null,
+  "partner_application_timestamp": null,
+  "is_published": false,
+  "reasons_to_join": [
+    { "reason": "Alien", "emoji_id": null, "emoji_name": "👽" },
+    { "reason": "Alien", "emoji_id": null, "emoji_name": "👽" },
+    { "reason": "Alien", "emoji_id": null, "emoji_name": "👽" },
+    { "reason": "Alien", "emoji_id": null, "emoji_name": "👽" }
```
Comment (splatterxl) [inline pages/resources/discovery.mdx:255]: Include a custom emoji here just for example completeness 😛

**Review by splatterxl — CHANGES_REQUESTED** [review]

File: pages/resources/discovery.mdx (hunk 2 above)
Comment (dolfies) [reply pages/resources/discovery.mdx:15]: Is it a client? I think that terminology is off.

File: pages/resources/discovery.mdx (hunk 3 above)
Comment (dolfies) [reply pages/resources/discovery.mdx:48]: I didn't design the API 😭
I don't think it's removed that quickly, and the client filters out auto-removed guilds when searching: `query=test&filters=auto_removed%3Afalse+AND+approximate_presence_count%3E+0+AND+approximate_member_count%3E200&optionalFilters=%5B%22preferred_locale%3A+en-US%22%5D&length=12&offset=0&restrictSearchableAttributes=%5B%22name%22%2C%22description%22%2C%22keywords%22%2C%22categories.name%22%2C%22categories.name_localizations.en-US%22%2C%22primary_category.name%22%2C%22primary_category.name_localizations.en-US%22%2C%22vanity_url_code%22%5D`. Also, there's a grace period before the guild is removed.

File: pages/resources/discovery.mdx (hunk 5 above)
Comment (dolfies) [reply pages/resources/discovery.mdx:152]: This would introduce inconsistency.

File: pages/resources/discovery.mdx (hunk 1 above)
Comment (dolfies) [reply pages/resources/discovery.mdx:9]: I don't think its worth maintaining a list for that.

**Review by arHSM — APPROVED** [review]

**Review by Dziurwa14 — APPROVED** [review]

**Review by splatterxl — APPROVED** [review]

**Pull request merged**